app/helpers/assets_helper.rb

```diff
@@ -1,23 +1,40 @@
 module AssetsHelper
-  def admin_style_tag(entrypoint_file_name)
-    compiled_css_file_path =
-      JSON.parse(File.read('public/vite-admin/.vite/manifest.json')).
-        dig("admin_entrypoints/#{entrypoint_file_name}", 'file')
-
-    stylesheet_link_tag("/vite-admin/#{compiled_css_file_path}")
-  end
+  def admin_ts_tag(entrypoint_name)
+    # rubocop:disable Layout/LineLength
+    # NOTE: To have an auto-reloading dev experience (though no HMR), launch Vite with:
+    #   VITE_RUBY_ENTRYPOINTS_DIR=admin_entrypoints VITE_RUBY_PUBLIC_OUTPUT_DIR=vite-admin ./node_modules/.bin/vite dev --force
+    # and launch a Rails server with:
+    #   VITE_RUBY_ENTRYPOINTS_DIR=admin_entrypoints VITE_RUBY_PUBLIC_OUTPUT_DIR=vite-admin bin/rails server
+    # rubocop:enable Layout/LineLength
+    if (
+      Rails.env.development? &&
+        ENV.fetch('VITE_RUBY_ENTRYPOINTS_DIR', nil) == 'admin_entrypoints' &&
+        ENV.fetch('VITE_RUBY_PUBLIC_OUTPUT_DIR', nil) == 'vite-admin' &&
+        ((Faraday.get(
+          "http://localhost:3036/vite-admin/admin_entrypoints/#{entrypoint_name}.ts",
+        ).status rescue nil) == 200)
+    )
+      vite_typescript_tag(entrypoint_name, defer: 'defer', crossorigin: nil)
```

Comment:
Before, I think maybe we never had any sort of fast-reloading dev environment? Or, if we did, I don't think that I knew how to use it. This aims to provide such an environment. We still won't have HMR, but at least we'll have full page reloading automatically triggered whenever a relevant file changes.